pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -713,13 +713,15 @@ func (ue *unsupportedFieldError) Error() string {
 
 // IsInvalidNameArg checks whether the name contains invalid characters
 func IsInvalidNameArg(name string) bool {
-	re := regexp.MustCompile(`[^a-zA-Z0-9\-]+`)
-	return re.MatchString(name)
+	startsWithNonAlphanumericRe := regexp.MustCompile(`^[^A-Za-z0-9]`)
+	containsUnsupportedCharacterRe := regexp.MustCompile(`[^A-Za-z0-9\-_]+`)
+
+	return startsWithNonAlphanumericRe.MatchString(name) || containsUnsupportedCharacterRe.MatchString(name)
```

Comment:
why do we need two regexes? Can we not use the original one 
```re := regexp.MustCompile(`[^a-zA-Z0-9\-]+`)```
to check and allow `underscores` ?
For ref see here https://github.com/weaveworks/eksctl/pull/3942

```suggestion
re := regexp.MustCompile(`[^a-zA-Z0-9\-_]+`)
	return re.MatchString(name)
```

Reply:
This regex doesn't reject names that start with `-` or `_`, but yeah I'll combine the two.